### PR TITLE
SDK-667 : Limit VCS info to 64kb

### PR DESCRIPTION
**Pull request by TamsilAmani**
Limiting the total size of VCS info to 64 Kb.

JIRA: https://browserstack.atlassian.net/browse/SDK-667

### CODE REVIEW

File: bin/helpers/helper.js [hunk 1]
```diff
@@ -387,3 +396,51 @@ exports.getSupportFiles = (bsConfig, isA11y) => {
...
+
+      return truncatedString;
+    }
+  } catch (error) {
+    exports.debug(`Error while truncating field, nothing was truncated here: ${error}`, true, error);
+    logger.debug(`Error while truncating field, nothing was truncated here: ${error}`, true, error);
+  }
+
+  return field;
+};
```
Comment (sauravdas1997) [inline bin/helpers/helper.js]: EOF

File: bin/helpers/helper.js [hunk 2]
```diff
@@ -387,3 +396,51 @@ exports.getSupportFiles = (bsConfig, isA11y) => {
...
+    const lenOfFieldBufferObj = fieldBufferObj.length;
+    const finalLen = Math.round(lenOfFieldBufferObj - (truncateSizeInKb * 1024) - (bufferSizeInBytes));
+    if (finalLen > 0) {
+      const truncatedString = fieldBufferObj.subarray(0, finalLen).toString() + GIT_META_DATA_TRUNCATED;
+
+      return truncatedString;
+    }
+  } catch (error) {
+    exports.debug(`Error while truncating field, nothing was truncated here: ${error}`, true, error);
+    logger.debug(`Error while truncating field, nothing was truncated here: ${error}`, true, error);
```
Comment (sauravdas1997) [inline bin/helpers/helper.js]: why are we logging the same line twice?

Comment (TamsilAmani) [reply bin/helpers/helper.js]: We can remove `exports.debug`

**Review by sauravdas1997 — APPROVED** [review]

File: bin/helpers/helper.js [hunk 3]
```diff
@@ -387,3 +396,49 @@ exports.getSupportFiles = (bsConfig, isA11y) => {
...
   };
 }
+
+exports.checkAndTruncateVCSInfo = (gitMetaData) => {
+  const gitMetaDataSizeInBytes = exports.getSizeOfJsonObjectInBytes(gitMetaData);
+
+  if (gitMetaDataSizeInBytes && gitMetaDataSizeInBytes > MAX_GIT_META_DATA_SIZE_IN_BYTES) {
+    const truncateSize = gitMetaDataSizeInBytes - MAX_GIT_META_DATA_SIZE_IN_BYTES;
+    const truncatedCommitMessage = exports.truncateString(gitMetaData.commit_message, truncateSize);
+    gitMetaData.commit_message = truncatedCommitMessage;
```
Comment (07souravkunda) [inline bin/helpers/helper.js]:
```suggestion
    gitMetaData.commit_message = exports.truncateString(gitMetaData.commit_message, truncateSize);;
```

**Review by 07souravkunda — CHANGES_REQUESTED** [review]

**Review by 07souravkunda — APPROVED** [review]

**Review by asambstack — APPROVED** [review]

**Pull request merged**